.github/workflows/build-tests.yml

```
@@ -0,0 +1,51 @@
+name: Build-Testing
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
```

Run by default in IOT folder. Then you don't need to set working dir for each step

```suggestion
    runs-on: ubuntu-24.04
    defaults:
      run:
        working-directory: IOT
```